tests/general/MementoBearTest.py

```diff
@@ -0,0 +1,306 @@
+
+    resp = requests.Response()
+    resp.raw = io.BytesIO()
+    resp.request = request
+    resp.status_code = status_code
+    return resp
+
+
+# Load memento headers file
+with open(get_testfile_path('headers1.json')) as headers1:
+    with open(get_testfile_path('headers2.json')) as headers2:
+        memento_headers = [json.load(headers1),
+                           json.load(headers2)]
+
+
+def memento_archive_status_mock(m, url, is_archived=True, override_head=True):
+    """
+    Mock the memento result for the given url.
+
+    :param m:             A `request_mock.Mocker()` instance.
+    :param url:           The ``url`` to be mocked.
+    :param is_archived:   True means the url will be marked as
+                          archived.
+    :param override_head: True means the url's head request will be overriden
+                          by this method.
+    """
+    if override_head:
+        m.head(url, headers={})
+    if is_archived:
+        m.head(
+            'http://timetravel.mementoweb.org/timegate/%s' % url,
+            headers=memento_headers[0],
+            status_code=302)
+        m.head(
+            'http://web.archive.org/web/20170421070937/'
+            'https://www.google.com/',
+            headers=memento_headers[1],
+            status_code=200)
+    else:
+        m.head(
+            'http://timetravel.mementoweb.org/timegate/%s' % url,
+            status_code=404)
+
+
+def generate_redirects(m, url, redirect_length=1):
+    """
+    Generate redirects path for the given url.
+
+    :param m:               A `request_mock.Mocker()` instance.
+    :param url:             The ``url`` to be redirected.
+    :param redirect_length: How many redirects will be generated.
+    """
+    m.head(
+        url,
+        headers={'Location': '%s/1' % url},
+        status_code=302)
+
+    if redirect_length > 1:
+        for i in range(redirect_length - 1):
+            m.head(
+                '%s/%s' % (url, str(i + 1)),
+                headers={'Location': '%s/%s' % (url, str(i + 2))},
+                status_code=302
+            )
+
+        m.head('%s/%s' % (url, str(redirect_length)),
+               status_code=200)
+    else:
+        m.head('%s/1' % url, status_code=200)
+
+
+class MementoBearTest(LocalBearTestHelper):
+    """
+    The tests are mocked (don't actually connect to internet) and
+    return the int conversion of the last three chars of
+    the URL as status code.
+
+    Check ``custom matcher`` for more info on implementation.
+    """
+
+    def assertResultCount(self, test_file, expected_num):
+        MementoBear.check_prerequisites = lambda *args: True
+        uut = MementoBear(self.section, Queue())
+        for line, num in zip(test_file, expected_num):
+            outputs = list(uut.run('testline', [line]))
+            self.assertEqual(num, len(outputs))
+
+    def setUp(self):
+        self.section = Section('')
+        MementoBear.check_prerequisites = lambda *args: True
```

Comment:
you need to add a `tearDown` method, which restores the real `MementoBear.check_prerequisites`